package/MDAnalysis/analysis/backends.py

```diff
@@ -0,0 +1,182 @@
+"""Analysis backends --- :mod:`MDAnalysis.analysis.backends`
+
+Provides :class:`BackendBase` base class to implement custom
+backends for :meth:`MDAnalysis.analysis.base.AnalysisBase.run()` and its
+subclasses. Two built-in backend classes classes are provided:
+:class:`BackendMultiprocessing` that supports parallelization via standard
+python ``multiprocessing`` module, and :class:`BackendDask`, that uses the same
+process-based parallelization as :class:`BackendMultiprocessing`, but different
+serialization algorithm.
+==============================================================
+"""
+import warnings
+from typing import Callable
+from MDAnalysis.lib.util import is_installed
+
+
+class BackendBase:
+    """Base class for backend implementation. Initializes an instance and performs
+    checks for its validity, such as n_workers and possibly other ones.
+
+    Parameters
+    ----------
+    n_workers : int
+        number of workers (usually, processes) over which the work is split
+
+    Examples
+    --------
+    .. code-block:: python
+    
+        from MDAnalysis.analysis.backends import BackendBase
+        class ThreadsBackend(BackendBase):
+            def apply(self, func, computations):
+                from multiprocessing.dummy import Pool
+
+                with Pool(processes=self.n_workers) as pool:
+                    results = pool.map(func, computations)
+                return results
+        from MDAnalysis.analysis.rms import RMSD
+        R = RMSD(...) # setup the run
+        n_workers = 2
+        backend = ThreadsBackend(n_workers=n_workers)
+        R.run(backend=backend)
```

Comment:
```suggestion
        R.run(backend=backend, unsupported_backend=True)
```
Perhaps also add a note to the docstring to mention that `unsupported_backend` must be `True` when using a user-defined backend